src/drivers/sdl/lv_sdl_window.c

```diff
@@ -209,8 +207,45 @@ static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_m
 #if LV_USE_DRAW_SDL == 0
     lv_sdl_window_t * dsc = lv_display_get_driver_data(disp);
     lv_color_format_t cf = lv_display_get_color_format(disp);
+    static uint32_t * argb_px_map = NULL;
 
     if(sdl_render_mode() == LV_DISPLAY_RENDER_MODE_PARTIAL) {
+
+        /*Update values in a special OLED I1 --> ARGB8888 case
+          We render everything in I1, but display it in ARGB8888*/
+        if(cf == LV_COLOR_FORMAT_I1) {
+            uint32_t argb_px_map_size;
+            /*I1 uses 1 bit wide pixels, ARGB8888 uses 4 byte wide pixels*/
+            cf = LV_COLOR_FORMAT_ARGB8888;
+            argb_px_map_size = lv_area_get_size(area) * 4;
+            argb_px_map = realloc(argb_px_map, argb_px_map_size);
+            if(argb_px_map == NULL) {
+                LV_LOG_ERROR("Malloc failed!\n");
+            }
```

Comment:
early return to avoid the `else` branch
```suggestion
            LV_ASSERT_MALLOC(argb_px_map);
            if(argb_px_map == NULL) {
                LV_LOG_ERROR("Malloc failed!\n");
                lv_display_flush_ready(display);
                return; 
            }
```

Reply:
ack